libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -61,6 +53,39 @@ class MapboxDirectionsSession(
         })
     }
 
+    override fun requestFasterRoute(
+        currentLocation: Location,
+        fasterRoutesRequestCallback: FasterRouteRequestCallback
+    ) {
+        if (routes.isEmpty()) {
+            return
+        }
+        routeOptions?.let { options ->
+            val origin = Point.fromLngLat(currentLocation.longitude, currentLocation.latitude)
+            val coordinates = options.coordinates().apply {
+                removeAt(0)
+                add(0, origin)
+            }
+            options.toBuilder().coordinates(coordinates)
+            router.getRoute(options, object : Router.Callback {
+                override fun onResponse(routes: List<DirectionsRoute>) {
+                    val isRouteFaster = fasterRoutesRequestCallback.onRouteReady(routes[0])
```

Comment:
I think `DirectionsSession` should just deliver the route to the client (`MapboxNavigation`) and that client should determine if the route is faster and notify the subscribers. We don't need to cycle back to the `DirectionsSession`.
This basically means that `FasterRouteCallback` could be maintained and invoked from `MapboxNavigation` instead.

Reply:
If you look at this https://github.com/mapbox/mapbox-navigation-android/pull/2455
> ```
> private fun requestFasterRoute(routeOptions: RouteOptions) {
>        if (routes.isEmpty()) {
>            return
>        }
>        router.getRoute(routeOptions, object : Router.Callback {
>            override fun onResponse(routes: List<DirectionsRoute>) {
>                val route = routes[0]
>                if (isRouteFaster(route)) {
>                    fasterRouteObservers.forEach { it.onFasterRouteAvailable(route) }
>                }
>            }
>
>            override fun onFailure(throwable: Throwable) {
>                // do nothing
>            }
>
>            override fun onCanceled() {
>                // do nothing
>            }
>        })
>    }
> ```

```
if (isRouteFaster(route)) {
       fasterRouteObservers.forEach { it.onFasterRouteAvailable(route) }
}
```

I have a `fasterRouteObserver` to which the end user subscribes to. This needs access to faster route. Hence, this is done this way

Reply:
Refs https://github.com/mapbox/mapbox-navigation-android/pull/2455/files#r379006808.